src/mem/chunkmap.h

```diff
@@ -172,22 +172,12 @@ namespace snmalloc
   struct DefaultChunkMap
   {
     /**
-     * Get the pagemap entry corresponding to a specific address.
-     *
-     * Despite the type, the return value is an enum ChunkMapSuperslabKind
-     * or one of the reserved values described therewith.
-     */
-    static uint8_t get(address_t p)
-    {
-      return PagemapProvider::pagemap().get(p);
-    }
-
-    /**
-     * Get the pagemap entry corresponding to a specific address.
+     * Get the pagemap entry corresponding to a specific address inside
+     * a ReturnPtr wrapper.
      */
-    static uint8_t get(void* p)
+    static uint8_t get(ReturnPtr p)
     {
-      return get(address_cast(p));
+      return get(address_cast(p.unsafe_return_ptr));
```

Comment:
I think this might be cleaner if `ReturnPtr` defined an `address_cast` member, then we're not requiring access to the raw pointer.